invenio_rdm_records/resources/serializers/iiif/schema.py

```diff
@@ -209,3 +214,17 @@ def sortcanvases(self, manifest, many, **kwargs):
         """
         # manifest["sequences"][0]["canvases"].sort(key=lambda x: x["@id"])
         return manifest
+
+    @pre_dump
+    def add_iiif_links(self, data, many):
+        """Add IIIF links to each files entry."""
+        if "entries" in data["files"]:
+            for file in data["files"]["entries"].values():
+                if "links" not in file:
+                    links_tpl = current_rdm_records_service.files.file_links_item_tpl(
```

Comment:
Comment: It does not seem right to use a service in here - I could be wrong, but I'd like to check first.

Reply:
Why does is not seem right? Is it because we have service logic in the presentation layer?

Reply:
It's because link generation is the responsibility of the service layer, and the service ought to pass all needed information to the serialise. Serializer's should focus only on transformation of service result into something a client understands.